### DISCUSSION TITLE
toggle/offset hover option for quick restart

### DISCUSSION THREAD

**Discussion by Communified** [ppy/osu#30710] [Ideas]
I understand the desire to have quick restart be faster and more responsive than before, but removing the option to adjust offset when quick restarting is more annoying than beneficial. Even without it being the default, it would be nice to have a toggle to go back to the mechanics that were there before, as most of the people playing lazer were already used to them, and as someone who often needs offset for beatmaps, it's a major struggle when I restart and then realize I can no longer make the adjustment like I could before, and then when manually restarting the offset data is gone, so I have to basically put in other useless attempt. 

I think the cleanest way if you don't want a toggle to get the best of both worlds is to keep it how it is, but if your cursor is hovering over the far right edge/border of the screen, allow the same offset/adjustment menu to pop up until you are no longer hovering in the area, incase you decide you WANT to make changes. That way you can still have fast beatmap loading but if you know you need to adjust offset, you can position your cursor to allow for the menu to appear.

**Comment by lemoncove** [comment]
In addition (or alternatively), the last-play offsets should not be overwritten if a new play is set which is too short to calculate an offset from.

As above, with the current behavior, I may quick-restart expecting to be able to change the offset based on my previous play, but the adjustment menu doesn't appear and instead the new play overwrites the previous stored offset even if I exit immediately.

A QoL bonus would be to allow maps to remember their last-play offsets even after other maps are played, so the offsets can be adjusted later. For example, when playing new maps in a multiplayer lobby, there's no opportunity to go back and set the offsets for next time after a play, but that may be desirable once the lobby ends.

**Comment by Electfried** [comment]
No kidding! Playing older beatmaps and having to press esc and click "Retry" to adjust offsets is making me go nuts! A simple solution to this would be to allow people to just press a key to set the local offset to whatever the game automatically calculated after a play. For example, I play a 2008 ranked beatmap that has a bad beatmap offset. I play it and do a quick restart; then I press whatever key is bound to adjust the local offset to what lazer calculated and boom simple fix. I feel like this would be easy to implement? I'm not no dev so I do not know how hard that would be.

Also, local offsets should be in the GUI to place on your gameplay field when doing skinning.

**Comment by cerfeque** [comment]
Everything worked fine until this patch. I wish developers just add the "quick restart" into the options so those who needs it can enable it. I personally dont mind waiting additional 0.5 seconds of loading time.

With the previous patch, when they gave us quick restart, i tauhgt myself to just press escape, go to the map selector and choose the same map again to see my offset, it was faster than other options. Now the game just forgets my offset graph when i go to the map select and choose the same map again. The only way to see my offset now is to finish the map. And after you restart you cant adjust it, game will forget it.
This is the only thing i see now, i already forgot how my offset pyramid looks... Game worked perfectly, this is the downgrade. 
![{14A0F563-12F7-4D9A-ADEB-29FB5E69069C}](https://github.com/user-attachments/assets/ef8b7e73-b0d1-4e95-aa55-e86b3af89a12)

**Comment by BDSuther** [comment]
If this ever gets looked at it would also be nice to stay paused indefinitely while the cursor is inside one of these UI settings elements when the map starts. Nothing is more annoying when I am trying to understand the audio offset graph or even just figure out what the audio offset is and then the map just starts playing.

I don't want to have to wiggle my mouse in the settings UI element to keep it from starting the map. In osu! classic, you could just open the settings at the bottom and leave the mouse there for 10 years and it would stay paused before starting the map.
